backend/books/views.py

```diff
@@ -0,0 +1,86 @@
+from django.core import serializers
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.generics import ListCreateAPIView, RetrieveUpdateDestroyAPIView
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated, IsAdminUser, AllowAny
+import re
+from .isbn import ISBNSearch
+from .serializers import BookSerializer
+from .models import Book, Author, Genre
+from pprint import pprint
+
+class ISBNSearchView(APIView):
+    permissions_classes = [IsAuthenticated]
+    search = ISBNSearch()
+
+    def post(self, request):
+        req_dict = request.data
+        if "isbns" in req_dict:
+            isbns = re.split("\s?[, ]\s?", req_dict['isbns'].strip())
```

Comment:
Might be good to add an explanation of what this is doing since its a bit cryptic.

Reply:
added comments and refactored code